### PR TITLE
Enforce constraints on usernames

### PR DESCRIPTION
**Pull request by zachmullen**

### CODE REVIEW

File: dkc/core/forms/signup.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import re
+
+from allauth.account.forms import SignupForm as SignupFormBase
+from django.core.exceptions import ValidationError
+
+username_regex = re.compile(r'^[a-z\d](?:[a-z\d]|-(?=[a-z\d])){0,49}$', flags=re.IGNORECASE)
+
+
+class SignupForm(SignupFormBase):
+    def clean_username(self):
```
Comment (brianhelba) [inline dkc/core/forms/signup.py:10]: I think this is a more appropriate place to run this logic: https://github.com/pennersr/django-allauth/blob/701d8ff905535e4cf79b6769ce9f60c7bed978ba/allauth/account/adapter.py#L249

We should be able to just add a callback to `ACCOUNT_USERNAME_VALIDATORS` ( https://django-allauth.readthedocs.io/en/latest/configuration.html ).

Comment (zachmullen) [reply dkc/core/forms/signup.py:10]: Excellent, I don't think @jbeezley or I were aware of that. I will change it.

Comment (zachmullen) [reply dkc/core/forms/signup.py:10]: @brianhelba I've converted it to use the new approach, but I'm not sure what the best way to test it is. Any thoughts?

Comment (zachmullen) [reply dkc/core/forms/signup.py:10]: Actually it turns out I can use the same approach and just use the out-of-the-box SignupForm from allauth.

**Comment by zachmullen** [comment]
@brianhelba this is now ready for another review pass.

File: dkc/settings.py [hunk 2]
```diff
@@ -10,6 +11,22 @@
     ProductionBaseConfiguration,
     TestingBaseConfiguration,
 )
+from django.core.validators import MaxLengthValidator, RegexValidator
+
+
+class UsernameRegexValidator(RegexValidator):
+    regex = r'^[a-z\d](?:[a-z\d]|-(?=[a-z\d]))*$'
```
Comment (brianhelba) [inline dkc/settings.py:18]: Why subclass instead of passing these as arguments to the constructor?

Comment (zachmullen) [reply dkc/settings.py:18]: I just copied how [these](https://github.com/django/django/blob/master/django/contrib/auth/validators.py) were done. I have no preference, I'm happy to change it if you want.

**Review by brianhelba — APPROVED** [review]

**Pull request merged**